inst/include/child_model_simulation.hpp

```diff
@@ -6,6 +6,65 @@ namespace leapfrog {
 
 namespace internal {
 
+template<HivAgeStratification S, typename real_type>
+void run_paed_ageing(int time_step,
+                     const Parameters<real_type> &pars,
+                     const State<S, real_type> &state_curr,
+                     State<S, real_type> &state_next,
+                     IntermediateData<S, real_type> &intermediate) {
+  const auto demog = pars.demography;
+  const auto children = pars.children;
+  
+  constexpr auto ss = StateSpace<S>();
+  for(int g = 0; g < ss.num_genders; ++g){
+    //less than 5 because there is a cd4 transition between ages 4 and 5
+    for(int ha = 1; ha < 5; ++ha) {
+      for(int hm = 0; hm < ss.hC1_disease_stages; ++hm){
+        for(int cat = 0 ; cat < ss.hTM; ++cat){
+          state_next.hc_hiv_pop(hm, cat, ha, g) += state_curr.hc_hiv_pop(hm, cat, ha-1, g) * demog.survival(ha, g, time_step);
+          
+        }
+        for(int dur = 0; dur < ss.treatment_stages; ++dur){
+          state_next.hc_art_pop(dur, hm, ha, g) += state_curr.hc_art_pop(dur, hm, ha-1, g) * demog.survival(ha, g, time_step);
+        }
+        
+      }
+    }
+  }
+  
+  for(int g = 0; g < ss.num_genders; ++g){
+    for(int hm = 0; hm < ss.hC1_disease_stages; ++hm){
+      for(int hm_alt = 0; hm_alt < ss.hC2_disease_stages; ++hm_alt){
+        for(int cat = 0 ; cat < ss.hTM; ++cat){
+          state_next.hc_hiv_pop(hm_alt, cat, 5, g) +=  state_curr.hc_hiv_pop(hm, cat, 4, g) * demog.survival(5, g, time_step) * children.hc_cd4_transition(hm_alt, hm);
+        }
+        for(int dur = 0; dur < ss.treatment_stages; ++dur){
+          state_next.hc_art_pop(dur, hm_alt, 5, g) += state_curr.hc_art_pop(dur, hm, 4, g) * demog.survival(5, g, time_step) * children.hc_cd4_transition(hm_alt, hm);
+        }
+      }
+    }
+  }
+  
+  for(int g = 0; g < ss.num_genders; ++g){
+    //thought it was ss.fertility_first_age_group but maybe not? ASK ROB
+    for(int ha = 6; ha < 15; ++ha) {
```

Comment:
Why does this start at 6 and end at 15? Can we relate this to hC1_disease_stages or hC2_disease_stages at all?

Reply:
Ah actually I think I can see what is happening, we have slightly different behaviour for 1-4, 5 and 6-14

Reply:
It looks like this is doing the same thing for 1-4 and 6 - 14 but slightly differently for 5. Could add a function for the duplicated code on 1-4 and 6-14 but up to you.

Reply:
I think leaving it outside of a function allows you to see more why the aging process is done like this? So I'm more inclined to keep it separate.